src/closed.jl

```diff
@@ -0,0 +1,44 @@
+"""
+Defines the ClosedSet type.
+
+A ClosedSet(left, right) is an interval set where both bounds are closed i.e. the range is [left, right]
+"""
+immutable ClosedSet{T}
+	left::T
+	right::T
+end
+
+function ClosedSet(left, right)
+	(I_left, I_right) = promote(left, right)
+	ClosedSet(I_left, I_right)
+end
+
+show(io::IO, I::ClosedSet) = print(io, "[", I.left, ",", I.right, "]")
```

Comment:
Oh, and this should probably print as `left..right` since that's how they will be constructed (that way you can copy/paste output into the REPL and have it work).